cognite/neat/rules/_analysis/_information_rules.py

```diff
@@ -0,0 +1,341 @@
+        properties = list(itertools.chain.from_iterable(class_property_pairs.values()))
+
+        for property_ in properties:
+            if property_.type_ == EntityTypes.object_property:
+                new_row = pd.Series(
+                    {
+                        "source_class": property_.class_,
+                        "connecting_property": property_.property_,
+                        "target_class": property_.value_type,
+                        "max_occurrence": property_.max_count,
+                    }
+                )
+                class_linkage = pd.concat([class_linkage, new_row.to_frame().T], ignore_index=True)
+
+        class_linkage.drop_duplicates(inplace=True)
+        class_linkage = class_linkage[["source_class", "connecting_property", "target_class", "max_occurrence"]]
+
+        return class_linkage
+
+    def connected_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
+        """Return a set of classes that are connected to other classes.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that are connected to other classes
+        """
+        class_linkage = self.class_linkage(consider_inheritance)
+        return set(class_linkage.source_class.values).union(set(class_linkage.target_class.values))
+
+    def defined_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
+        """Returns classes that have properties defined for them in the data model.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that have been defined in the data model
+        """
+        class_property_pairs = self.classes_with_properties(consider_inheritance)
+        properties = list(itertools.chain.from_iterable(class_property_pairs.values()))
+
+        return {property.class_ for property in properties}
+
+    def disconnected_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
+        """Return a set of classes that are disconnected (i.e. isolated) from other classes.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that are disconnected from other classes
+        """
+        return self.defined_classes(consider_inheritance) - self.connected_classes(consider_inheritance)
+
+    def symmetrically_connected_classes(
+        self, consider_inheritance: bool = False
+    ) -> set[tuple[ClassEntity, ClassEntity]]:
+        """Returns a set of pairs of symmetrically linked classes.
+
```

Comment:
A sentence of what a symmetrically connected class is would be helpful.

Reply:
Good catch ! Added